tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import pytest
+import random
+from ocs_ci.framework.testlib import (
+    tier4, ManageTest, aws_platform_required, ignore_leftovers)
+from ocs_ci.ocs import machine
+from ocs_ci.ocs import constants
+from ocs_ci.ocs import ocp
+from ocs_ci.ocs.resources import pod
+from tests.helpers import get_worker_nodes
+from tests.helpers import wait_for_resource_state
+from tests.sanity_helpers import Sanity
+
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@ignore_leftovers
+@aws_platform_required
+class Testknip678(ManageTest):
+    """
+    Knip-678 Automated recovery from failed nodes
+    """
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    def test_knip678_proactive(self, pvc_factory, pod_factory):
+        """
+        Knip-678 Automated recovery from failed nodes
+        Proactive case - IPI
+        """
+        # Creating resources (pools, storageclasses, PVCs,
+        # pods - both CephFS and RBD), run IO
+        self.sanity_helpers.create_resources(pvc_factory, pod_factory)
+
+        # Get the osd associated node name
+        osd_pods_obj = pod.get_osd_pods()
+        osd_node_name = pod.get_pod_node(random.choice(osd_pods_obj)).name
+        log.info(f"Selected OSD is {osd_node_name}")
+
+        # Get the machine name using the node name
+        machine_name = machine.get_machine_from_node_name(osd_node_name)
+        log.info(f"{osd_node_name} associated machine is {machine_name}")
+
+        # Get the machineset name using machine name
+        machineset_name = machine.get_machineset_from_machine_name(machine_name)
+        log.info(f"{osd_node_name} associated machineset is {machineset_name}")
+
+        # Get the initial nodes list
+        initial_nodes = get_worker_nodes()
+        log.info(f"Initial current available nodes are {initial_nodes}")
+
+        # get machineset replica count
+        machineset_replica_count = machine.get_replica_count(machineset_name)
+        log.info(
+            f"{machineset_name} has replica count: {machineset_replica_count}"
+        )
+
+        # Increase its replica count
+        machine.add_node(machineset_name, count=machineset_replica_count + 1)
```

Comment:
Since this functionality is going to be used in multiple tests, I suggest you create a function in some common file, that does add and label node

Reply:
Done. Created a new function with name "add_new_node_and_label_it" in node.py